packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -564,37 +578,90 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
-    private tooltip(x: number, y: number): void {
+    private closestPoint(array: any, point: any, xRatio: number, yRatio: number, chartHeight: number): number | undefined {
+        const hypos: any[] = [];
+
+        array.forEach((ar: any) => {
+            // Convert x values to screen pixels
+            const pointX = point.x * xRatio;
```

Comment:
pointX and pointY are constant in this method so they can be calculated before the loop.

Reply:
Done.